src/agent/solana/oracle.rs

```diff
@@ -384,24 +414,73 @@ impl Poller {
     }
 
     async fn poll_and_send(&mut self) -> Result<()> {
+        let fresh_data = self.poll().await?;
+
+        self.perm_price_tx
+            .send(fresh_data.perm_price_accounts.clone())
+            .await
+            .context("Updating permissioned price accounts for exporter")?;
+
         self.data_tx
-            .send(self.poll().await?)
+            .send(fresh_data)
             .await
-            .map_err(|_| anyhow!("failed to send data to oracle"))
+            .context("failed to send data to oracle")?;
+
+        Ok(())
     }
 
     async fn poll(&self) -> Result<Data> {
         let mapping_accounts = self
-            .fetch_mapping_accounts(self.mapping_account_key)
+            .fetch_mapping_accounts(self.key_store.mapping_key)
             .await?;
         let (product_accounts, price_accounts) = self
             .fetch_product_and_price_accounts(mapping_accounts.values())
             .await?;
 
+        let publish_keypair = if let Some(kp) = self.key_store.publish_keypair.as_ref() {
```

Comment:
needing the publisher keypair here feels inelegant. I think maybe a simpler way to do this would be to return the full mapping from price accounts -> permissioned publishers, then check which ones are permissioned in the exporter (where we already have this keypair loading logic)

Reply:
Done, we get a hashmap in exporter and find relevant entry for the publisher using just the single keypair request `if`.